### PR TITLE
Add refresh for file

### PR DESCRIPTION
**Pull request by romanbracinik**
Jira: BIG-226
KBC: https://github.com/keboola/connection/pull/4815

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

Text som este upravil, obrazky su len ilustracne ze to funguje :) 

Screenshot 2024-01-30 at 18 32 15
Screenshot 2024-01-30 at 18 32 22

### CODE REVIEW

**Comment by romanbracinik** [comment]
Implementaciu retry hodim do zvlast PR

File: tests/File/GcsFileTest.php [hunk 1]
```diff
@@ -49,6 +53,45 @@ public function setUp(): void
         );
     }
 
+    /**
+     * @dataProvider provideComponentsClientTypeBasedOnSuite
+     */
+    public function testUploadFileWithRefreshedCredentials(): void
```
Comment (zajca) [inline tests/File/GcsFileTest.php:59]: ten test mě padá v SOXu 
image
možeš to zkusit?

Comment (romanbracinik) [reply tests/File/GcsFileTest.php:59]: Skusim, skusal som to viac krat a nemal som problem.

Comment (romanbracinik) [reply tests/File/GcsFileTest.php:59]: Aha on sa ten test este pusta v SOX suite to som si nevsimol, pozriem sa na to. Dik

Comment (zajca) [reply tests/File/GcsFileTest.php:59]: jj tam bude problém že nepoužíváš branch clienta a snažíš sa zapsat do produkce jako developer.

Comment (romanbracinik) [reply tests/File/GcsFileTest.php:59]: Screenshot 2024-01-31 at 15 30 59
Dik fixol som to

**Review by zajca — APPROVED** [review]

**Pull request merged**